### PR TITLE
Fix the dataset merge, export on Machine B

### PR DESCRIPTION
**Pull request by JSabadin**
## Purpose

**Bug description:**
When a dataset created on GCS is pulled down to Machine B and used as a local dataset, attempting to merge it with another local dataset on Machine B produces an invalid dataset. Notably, merging two datasets directly on the cloud from different machines worked correctly prior to this.

This edge case has been addressed in this PR, and a corresponding test has been added.

Additionally, the export now includes metadata such as text for OCR.

## Specification

None / not applicable

## Dependencies & Potential Impact

None / not applicable

## Deployment Plan

None / not applicable

## Testing & Validation

None / not applicable

### CODE REVIEW

File: luxonis_ml/data/datasets/luxonis_dataset.py [hunk 1]
```diff
@@ -269,7 +269,10 @@ def _merge_metadata_with(self, other: "LuxonisDataset") -> None:
         self._write_metadata()
 
     def clone(
-        self, new_dataset_name: str, push_to_cloud: bool = True
+        self,
+        new_dataset_name: str,
+        push_to_cloud: bool = True,
+        team_id: str | None = None,
```
Comment (klemen1999) [inline luxonis_ml/data/datasets/luxonis_dataset.py:275]: Let's add it to CLI as well

**Review by klemen1999 — APPROVED** [review]

**Review by kozlov721 — APPROVED** [review]
